calyx-opt/src/passes/static_inliner.rs

```diff
@@ -0,0 +1,388 @@
+                par_group.borrow_mut().attributes = attributes.clone();
+                par_group
+            }
+            ir::StaticControl::If(ir::StaticIf {
+                port,
+                tbranch,
+                fbranch,
+                latency,
+                attributes,
+            }) => {
+                let if_group = builder.add_static_group("static_if", *latency);
+                let mut if_group_assigns: Vec<
+                    ir::Assignment<ir::StaticTiming>,
+                > = vec![];
+                structure!( builder;
+                    let cond = prim std_reg(port.borrow().width);
+                    let cond_wire = prim std_wire(port.borrow().width);
+                );
+                // build_cond_assigns makes assigns such that
+                // cond_wire.in can guard all of the tru branch assigns,
+                // and !cond_wire.in can guard all fo the false branch assigns x
+                let cond_assigns = StaticInliner::make_cond_assigns(
+                    &cond, &cond_wire, port, *latency, builder,
+                );
+                if_group_assigns.extend(cond_assigns.to_vec());
+                let tbranch_latency = tbranch.get_latency();
+                let fbranch_latency = fbranch.get_latency();
+                // turn tbranch into group and put assigns into tgroup_assigns
+                let tgroup =
+                    StaticInliner::inline_static_control(tbranch, builder);
+                assert_eq!(
+                    tbranch_latency,
+                    tgroup.borrow().get_latency(),
+                    "tru branch and tru branch group latency do not match"
+                );
+                let mut tgroup_assigns: Vec<ir::Assignment<ir::StaticTiming>> =
+                    tgroup.borrow_mut().assignments.clone();
+                // turn fgroup (if it exists) into group and put assigns into fgroup_assigns
+                let mut fgroup_assigns: Vec<ir::Assignment<ir::StaticTiming>> =
+                    match **fbranch {
+                        ir::StaticControl::Empty(_) => vec![],
+                        _ => {
+                            let fgroup = StaticInliner::inline_static_control(
+                                fbranch, builder,
+                            );
+                            assert_eq!(fbranch_latency, fgroup.borrow().get_latency(), "tru branch and tru branch group latency do not match");
+                            let fgroup_assigns: Vec<
+                                ir::Assignment<ir::StaticTiming>,
+                            > = fgroup.borrow_mut().assignments.clone();
+                            fgroup_assigns
+                        }
+                    };
+
+                // update trgoup_assigns to have guard %[0:tbranch_latency] in front of
+                // each assignment, and %[0:fbranch_latency] for fgroup_assigns
+                StaticInliner::update_assignment_timing(
```

Comment:
Seems like the assignments on the `tgroup` and `fgroup` are iterated over twice. Once to add the timing guard and once to add the conditional guard. We can probably do this in one pass?